main.cpp

```diff
@@ -0,0 +1,247 @@
+/* Stellar-CSO
+   A tool for converting ISO/XISO files to CSO (Compressed ISO)
+
+   Copyright 2018 David O'Rourke <david.orourke@gmail.com>
+   Copyright 2022 MakeMHz LLC <contact@makemhz.com>
+   Based on ciso from https://github.com/jamie/ciso
+*/
+
+#include <lz4frame.h>
+#include <lz4hc.h>
+
+#include <iostream>
+#include <ios>
+#include <filesystem>
+#include <fstream>
+#include <string>
+#include <vector>
+
+const uint32_t FATX_SIZE_LIMIT = 0xFFBF6000;
+
+const std::string CISO_MAGIC = "CISO";
+const uint32_t CISO_HEADER_SIZE = 0x18;
+const uint32_t CISO_BLOCK_SIZE = 0x800;
+const uint32_t CISO_PLAIN_BLOCK = 0x80000000;
+
+size_t image_offset = 0;
+
+void detect_iso_type(std::ifstream& f) {
+    try {
+	f.seekg(0x18310000, std::ios::beg);
+    } catch (std::exception const& e) {
+	std::cout << "File not big enough to be a Redump style image." << std::endl;
```

Comment:
nit: Why are we flushing on almost every cout? And there's some cerr's where we immediately exit, are we not trusting the OS?